### ISSUE TITLE
Do not rely on `sqlx` database connection in client integration tests

### ISSUE THREAD

**Issue by pohlm01** [OpenLEADR/openleadr-rs#53]
The integration tests for `event` and `program` in the VEN library rely on a DB created by the sqlx testing framework. This is not ideal, especially if we want to test the integration of our VEN library with a 3rd party VTN. Instead, the tests should rely on a running VTN instance and connect to a predefined URL.
This also requires a predefined set of credentials with associated roles in the VTN, which the tests can rely on. Currently, they are created from the [`test_user_credentials.sql`](https://github.com/OpenLEADR/openleadr-rs/blob/8ca55464b65f52b2b0d24ae6233f458f90495a10/fixtures/test_user_credentials.sql) and applied in the CI before test execution.
The tests must also run serially instead of in parallel, to avoid interference between them.

The tests for `resources` and `ven` do that already, though there might be room for improvement on how to design the tests exactly. Especially, a failing test may currently leave the VTN/DB in a different state than before.

**Comment by pohlm01** [comment]
@joshuathayer thanks for your first contribution in #69! Feel free to pick this up as you suggested yourself. I'm happy to help you with any questions that might arise.

**Comment by pohlm01** [comment]
Yes, you understand the description correctly. The reason is that we added the implementation and tests for `resource` and `ven` later than for `program` and `event` (`reports` are missing tests altogether, see #75). At that later movement, when we added `resource` and `ven` tests, we made sure not to depend on the DB connection in the client library. So you can definitely draw - possibly a lot of - inspiration from those implementations if you want.
We appreciate any improvement over the currently DB-dependent tests. Even if they don't feel perfect yet, we can always improve further later. I recommend opening a (draft) PR already in an early state.

**Comment by ryanshepps** [comment]
@pohlm01 Would you be open to using https://rust.testcontainers.org/? Otherwise I can continue off of https://github.com/OpenLEADR/openleadr-rs/pull/100.
